### PR TITLE
SNOW-850048 Hanlde of subdirectiores by GET from stage

### PR DESCRIPTION
**Pull request by sfc-gh-dstempniak**
Proper handling of remote path containing a sub-directory in GET command

### CODE REVIEW

File: Snowflake.Data.Tests/SFPutGetTest.cs [hunk 1]
```diff
@@ -2,418 +2,259 @@
...
                 {
-                    // Create temp table
-                    command.CommandText = createTable;
-                    command.ExecuteNonQuery();
-
-                    // Create temp stage
-                    command.CommandText = createStage;
-                    command.ExecuteNonQuery();
+                    // ********************************************************************
+                    // ** 1. PUT - upload file from local directory to the stage
```
Comment (sfc-gh-dprzybysz) [inline Snowflake.Data.Tests/SFPutGetTest.cs:160]: do we need such section comments? maybe we should create helper function to describe them?

File: Snowflake.Data.Tests/SFPutGetTest.cs [hunk 2]
```diff
@@ -2,418 +2,259 @@
...
+                    // ********************************************************************
+                    
+                    // Prepare PUT query - Windows user contains a '~' in the path which causes an error
+                    string putQuery = $"PUT file://{inputFilePath} {internalStage} AUTO_COMPRESS=" + (autoCompress ? "TRUE" : "FALSE");
```
Comment (sfc-gh-dprzybysz) [inline Snowflake.Data.Tests/SFPutGetTest.cs:164]: why `autoCompress ? "TRUE" : "FALSE"` is not part of interpolated string but concatenated separately?

File: Snowflake.Data.Tests/SFPutGetTest.cs [hunk 3]
```diff
@@ -2,418 +2,259 @@
...
+
+                    // Check file status
+                    Assert.AreEqual(UPLOADED,
+                        reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus));
+                    // Check source and destination compression type
+                    if (autoCompress)
+                    {
+                        Assert.AreEqual(compressionType, 
+                            reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.SourceCompressionType));
+                        Assert.AreEqual( compressionType,
```
Comment (sfc-gh-dprzybysz) [inline Snowflake.Data.Tests/SFPutGetTest.cs:179]: formatting here is inconsistent with statement above

File: Snowflake.Data.Tests/SFPutGetTest.cs [hunk 4]
```diff
@@ -2,418 +2,259 @@
...
+                            reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.DestinationCompressionType));
+                    }
+                    else
                     {
-                        // Check file status
-                        Assert.AreEqual(reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus), UPLOADED);
+                        Assert.AreEqual("none",
```
Comment (sfc-gh-dprzybysz) [inline Snowflake.Data.Tests/SFPutGetTest.cs:184]: `"none"` is kind of const? can we name it explicitly?

File: Snowflake.Data.Tests/SFPutGetTest.cs [hunk 5]
```diff
@@ -2,418 +2,259 @@
...
-
-                // Delete temp file
-                File.Delete(filePath);
-
-                conn.Close();
-                Assert.AreEqual(ConnectionState.Closed, conn.State);
             }
+
+            // Delete temp files
+            File.Delete(outputFilePath);
```
Comment (sfc-gh-dprzybysz) [inline Snowflake.Data.Tests/SFPutGetTest.cs:256]: shouldn't it be done in some tear down section?

File: Snowflake.Data/Core/FileTransfer/StorageClient/SFRemoteStorageUtil.cs [hunk 6]
```diff
@@ -467,7 +467,7 @@ internal static async Task DownloadOneFileAsync(SFFileMetadata fileMetadata, Can
             else
             {
                 var msg = "Unknown Error in downloading a file: " + fileMetadata.destFileName;
-                throw lastErr;
+                throw new Exception(msg);
```
Comment (sfc-gh-dprzybysz) [inline Snowflake.Data/Core/FileTransfer/StorageClient/SFRemoteStorageUtil.cs:470]: isn't that unexpected error from user perspective? or maybe we should pass cause to new exception?

**Review by sfc-gh-ext-simba-hx — APPROVED** [review]
LGTM

**Review by sfc-gh-pbulawa — APPROVED** [review]

**Pull request merged**